File: festivalFlow-be/src/main/java/com/sunvalley/festivalFlowbe/entity/LocationEntity.java
package com.sunvalley.festivalFlowbe.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "location", schema = "FestivalFlow")
public class LocationEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(name = "name", nullable = false)
  private String name;

  @Column(name = "description")
  private String description;

  @ManyToOne
  @JoinColumn(name = "day_id", referencedColumnName = "id")
  private DayEntity day;

  @Column(name = "adultsOnly", nullable = false)
  private boolean adultsOnly;

  @Column(name = "manager")
  private String manager;

  @Column(name = "image")
  private String image;

}

File: festivalFlow-fe/src/app/interfaces/LocationEntity.ts
import {Day} from "./DayEntity";

export interface Location {
  id?: number;
  name: string;
  description?: string | null;
  day: Day;
  adultsOnly?: boolean;
  manager?: string;
  image?: string;
}

export interface LocationClient {
  id?: number;
  name: string;
  description?: string | null;
  day: Day;
  adultsOnly?: boolean;
  manager?: string;
  image?: string;
  shiftAvailability: number;
}

